### PR TITLE
Remove usage of deprecated methods

### PR DESCRIPTION
**Pull request by amottier**
- Replace usage of `$doc.name` with `$doc.pageReference.name`.
- Replace usage of `$doc.realLanguage` with `$doc.realLocale.language`.

This should close https://jira.xwiki.org/browse/BLOG-137 and https://jira.xwiki.org/browse/BLOG-155.

### CODE REVIEW

File: application-blog-ui/src/main/resources/Blog/RssCode.xml [hunk 1]
```diff
@@ -163,7 +163,7 @@
       ## This is just a list of blog entries, which are detailed below.
       #foreach ($entryDoc in $entries)
         #if($xwiki.hasAccessLevel('view', ${entryDoc.fullName}))
-        &lt;rdf:li rdf:resource="$entryDoc.getExternalURL('view', "language=${entryDoc.realLanguage}")" /&gt;
+        &lt;rdf:li rdf:resource="$entryDoc.getExternalURL('view', "language=${entryDoc.realLocale.language}")" /&gt;
```
Comment (mflorea) [inline application-blog-ui/src/main/resources/Blog/RssCode.xml:166]:
```suggestion
        &lt;rdf:li rdf:resource="$entryDoc.getExternalURL('view', "language=${entryDoc.realLocale}")" /&gt;
```

I know the query string parameter is ``language`` but in practice it is used as a locale (you can pass for instance ``?language=fr_CA``).

**Review by mflorea — CHANGES_REQUESTED** [review]

File: application-blog-ui/src/main/resources/Blog/RssCode.xml [hunk 2]
```diff
@@ -194,7 +194,7 @@
       ## This is just a list of blog entries, which are detailed below.
       #foreach ($entryDoc in $entries)
         #if($xwiki.hasAccessLevel('view', ${entryDoc.fullName}))
-        &lt;rdf:li rdf:resource="$entryDoc.getExternalURL('view', "language=${entryDoc.realLanguage}")" /&gt;
+        &lt;rdf:li rdf:resource="$entryDoc.getExternalURL('view', "language=${entryDoc.realLocal.language}")" /&gt;
```
Comment (tmortagne) [inline application-blog-ui/src/main/resources/Blog/RssCode.xml:197]: @amottier don't forget this one ;)

File: application-blog-ui/src/main/resources/Blog/RssCode.xml [hunk 3]
```diff
@@ -229,7 +229,7 @@
       ## This is just a list of blog entries, which are detailed below.
       #foreach ($entryDoc in $entries)
         #if($xwiki.hasAccessLevel('view', ${entryDoc.fullName}))
-        &lt;rdf:li rdf:resource="$entryDoc.getExternalURL('view', "language=${entryDoc.realLanguage}")" /&gt;
+        &lt;rdf:li rdf:resource="$entryDoc.getExternalURL('view', "language=${entryDoc.realLocale.language}")" /&gt;
```
Comment (tmortagne) [inline application-blog-ui/src/main/resources/Blog/RssCode.xml:232]: And this one.

File: application-blog-ui/src/main/resources/Blog/RssCode.xml [hunk 4]
```diff
@@ -279,7 +279,7 @@
  * @param entryDoc The XDocument corresponding to the displayed blog entry.
  *###
 #macro(printBlogRssItem $entryDoc)
-  #set($entryUrl = $entryDoc.getExternalURL('view', "language=${entryDoc.realLanguage}"))
+  #set($entryUrl = $entryDoc.getExternalURL('view', "language=${entryDoc.realLocale.language}"))
```
Comment (tmortagne) [inline application-blog-ui/src/main/resources/Blog/RssCode.xml:282]: And finally that one.

**Review by tmortagne — CHANGES_REQUESTED** [review]

**Review by mflorea — APPROVED** [review]

File: application-blog-ui/src/main/resources/Blog/RssCode.xml (hunk 2 above)
Comment (amottier) [reply application-blog-ui/src/main/resources/Blog/RssCode.xml]: thanks @tmortagne and I also fix the typo `realLocal` → `realLocale`

File: application-blog-ui/src/main/resources/Blog/RssCode.xml (hunk 4 above)
Comment (amottier) [reply application-blog-ui/src/main/resources/Blog/RssCode.xml]: Done

File: application-blog-ui/src/main/resources/Blog/RssCode.xml (hunk 3 above)
Comment (amottier) [reply application-blog-ui/src/main/resources/Blog/RssCode.xml]: :heavy_check_mark:

**Comment by tmortagne** [comment]
Thanks for the contribution @amottier !

**Comment by amottier** [comment]
> Thanks for the contribution @amottier !

Your welcome! I'm a maniac of clean logs, so it push me to contribute :wink:

**Comment by amottier** [comment]
@tmortagne I was wondering if you now when a new release is planned to benefit from those changes? Thanks.

**Pull request merged**